src/plugins/camera/camera_impl.cpp

```diff
@@ -925,24 +929,59 @@ void CameraImpl::process_camera_image_captured(const mavlink_message_t& message)
         capture_info.is_success = (image_captured.capture_result == 1);
         capture_info.index = image_captured.image_index;
 
+        LogDebug() << "Received image_captured: " << capture_info.index;
+
         if (capture_info.is_success) {
             _status.photo_list.insert(std::make_pair(image_captured.image_index, capture_info));
         }
 
         _captured_request_cv.notify_all();
 
         std::lock_guard<std::mutex> lock(_capture_info.mutex);
-        if (_capture_info.last_advertised_image_index != -1 &&
-            _capture_info.last_advertised_image_index < capture_info.index &&
-            _capture_info.callback) {
+        if (_capture_info.last_advertised_image_index != -1) {
+            // Notify user if a new image has been captured.
+            if (_capture_info.last_advertised_image_index < capture_info.index) {
+                LogDebug() << "Call user for " << capture_info.index;
+                if (_capture_info.callback) {
+                    const auto temp_callback = _capture_info.callback;
+                    _parent->call_user_callback(
+                        [temp_callback, capture_info]() { temp_callback(capture_info); });
+                }
+            }
+
+            else if (auto it = _capture_info.missing_image_indices.find(capture_info.index); it != _capture_info.missing_image_indices.end()) {
+                LogDebug() << "Call user later for " << capture_info.index;
+                if (_capture_info.callback) {
+                    const auto temp_callback = _capture_info.callback;
+                    _parent->call_user_callback(
+                        [temp_callback, capture_info]() { temp_callback(capture_info); });
+                }
+                _capture_info.missing_image_indices.erase(it);
+            }
+
+            // Save captured indices that have been
```

Comment:
that have been... missed?
```suggestion
            // Save captured indices that have been missed
```